### PR TITLE
Fix #4648: Let the combiner combine with things other than cobblestone

### LINKED ISSUE #4648: Feature request: Allow non-cobblestone in the combiner
Currently, the combiner is hardcoded to only combine things with cobblestone, so it's not possible, for example, to add a recipe for it to combine dusts with netherrack to make nether ores. Can its recipe system be updated to support things other than cobblestone, so that other mods (and the upcoming CraftTweaker integration) can add these sorts of recipes?

### PR DESCRIPTION
**Pull request by josephcsible**
This lets us combine netherrack with quartz to make nether quartz ore now, and can be extended to combine any two arbitrary items. Note the following two (rather minor) breaking changes:
1. Cobblestone used to be consumed at the beginning of work. It's now consumed at the end. Any combiners that were running prior to this update will lose the single piece of cobblestone that they had consumed but not yet finished using.
2. Any CraftTweaker scripts that used the three-argument form of `mods.mekanism.combiner.addRecipe` (passing `<gas:liquidStone>` as the second argument) will break.

### CODE REVIEW

File: src/main/java/mekanism/common/recipe/machines/CombinerRecipe.java [hunk 1]
```diff
@@ -1,20 +1,26 @@
...
 	{
 		super(input, output);
 	}
 
+	public CombinerRecipe(ItemStack input, ItemStack extra, ItemStack output)
+	{
+		super(input, extra, output);
+	}
+
+	@Deprecated // TODO remove this method in MC 1.13
```
Comment (thiakil) [inline src/main/java/mekanism/common/recipe/machines/CombinerRecipe.java]: Why remove it? That would be a convenience method for the default.

**Comment by thiakil** [comment]
Iis there a particular reason behind a new recipe/TE/container/gui type that exists to only be extended by the combiner? The combiner already has its own `RecipeType`

File: src/main/java/mekanism/common/recipe/machines/CombinerRecipe.java (hunk 1 above)
Comment (josephcsible) [reply src/main/java/mekanism/common/recipe/machines/CombinerRecipe.java]: It seems rather arbitrary for cobblestone to be the default, and there's no way we'd add this method if we were building the combiner like this from scratch today.

**Comment by josephcsible** [comment]
I imagine it could be useful in future machines, and there's already precedent for it: ChanceMachineRecipe is only extended by SawmillRecipe.

File: src/main/java/mekanism/common/tile/prefab/TileEntityDoubleElectricMachine.java [hunk 2]
```diff
@@ -0,0 +1,287 @@
...
+	 *
+	 * @param soundPath - location of the sound effect
+	 * @param name - full name of this machine
+	 * @param maxEnergy - maximum amount of energy this machine can hold.
+	 * @param baseEnergyUsage - how much energy this machine uses per tick.
+	 * @param ticksRequired - how many ticks it takes to smelt an item.
+	 */
+	public TileEntityDoubleElectricMachine(String soundPath, String name, double maxEnergy, double baseEnergyUsage, int ticksRequired)
+	{
+		super(soundPath, name, maxEnergy, baseEnergyUsage, 4, ticksRequired, MekanismUtils.getResource(ResourceType.GUI, "guibasicmachine.png"));
```
Comment (josephcsible) [inline src/main/java/mekanism/common/tile/prefab/TileEntityDoubleElectricMachine.java]: On the combiner today, this is [guiadvancedmachine.png](https://github.com/aidancbrady/Mekanism/blob/89e10b1d9d0399ca821c30db3f416638522ee723/src/main/resources/assets/mekanism/gui/guiadvancedmachine.png), which has a bar between the cobblestone and dust slots that showed the liquid stone level. I'm not sure whether it'd be better to change that to [guibasicmachine.png](https://github.com/aidancbrady/Mekanism/blob/89e10b1d9d0399ca821c30db3f416638522ee723/src/main/resources/assets/mekanism/gui/guibasicmachine.png) as I did here, which replaces the bar with an arrow pointing up from the cobblestone to the dust, or to [guiblank.png](https://github.com/aidancbrady/Mekanism/blob/89e10b1d9d0399ca821c30db3f416638522ee723/src/main/resources/assets/mekanism/gui/guiblank.png), which would just remove the bar.

**Comment by josephcsible** [comment]
I also wonder whether we should make each ”lane” of the factories have their own cobblestone/netherrack/whatever slot, instead of all sharing one, so the same factory could do more than one kind of combining at once (although I don't think the infusion factory supports this either, so I'm not overly concerned right now).

File: src/main/java/mekanism/common/recipe/machines/CombinerRecipe.java (hunk 1 above)
Comment (thiakil) [reply src/main/java/mekanism/common/recipe/machines/CombinerRecipe.java]: > arbitrary 

I disagree, given that 99% of ores are made from cobble/stone

File: src/main/java/mekanism/common/tile/prefab/TileEntityDoubleElectricMachine.java (hunk 2 above)
Comment (thiakil) [reply src/main/java/mekanism/common/tile/prefab/TileEntityDoubleElectricMachine.java]: yeah `basicmachine` should be fine

**Comment by thiakil** [comment]
It would probably get too messy if you added a slot per lane. You will need to account for that specially if you're consuming the item upon completion of course

File: src/main/java/mekanism/common/recipe/machines/CombinerRecipe.java (hunk 1 above)
Comment (josephcsible) [reply src/main/java/mekanism/common/recipe/machines/CombinerRecipe.java]: We don't have convenience methods for any other recipes, though. It just doesn't seem natural to have one here. (But if this is really a big deal, I am willing to un-deprecate them to get this merged though.)

**Comment by josephcsible** [comment]
Okay, I think it's good for now with just the one slot. (And we can always change that later if we really want.)

File: src/main/java/mekanism/common/recipe/machines/CombinerRecipe.java (hunk 1 above)
Comment (thiakil) [reply src/main/java/mekanism/common/recipe/machines/CombinerRecipe.java]: I'm fine with you suggesting it, but putting it in without mentioning it comes across as "we must do this"

To me this constructor defines what the default is

Comment (josephcsible) [reply src/main/java/mekanism/common/recipe/machines/CombinerRecipe.java]: Fair point; I'll reword the comment a bit.

**Pull request merged**